src/librustc_metadata/creader.rs

```diff
@@ -854,21 +854,45 @@ impl<'a> CrateLoader<'a> {
                 return
             }
 
-            if !self.sess.crate_types.borrow().iter().all(|ct| {
-                match *ct {
-                    // Link the runtime
-                    config::CrateTypeExecutable => true,
-                    // This crate will be compiled with the required
-                    // instrumentation pass
-                    config::CrateTypeRlib => false,
-                    _ => {
-                        self.sess.err(&format!("Only executables and rlibs can be \
-                                                compiled with `-Z sanitizer`"));
-                        false
+            // The gnu linux target has better support for asan with multiple lib types.
```

Comment:
Ok that's fine, but can you list that in the comment here? I'm not sure the comment is correct as-is and I figured we could just make sure it's reflecting the reality we're sure exists